jiant/utils/utils.py

```diff
@@ -361,15 +401,40 @@ def get_elmo_mixing_weights(text_field_embedder, task=None):
     return params
 
 
-def get_batch_size(batch, keyword="input"):
+def format_output(obj, cuda_devices):
+    """
+    Format output based on whether model is using DataParallel or not. 
+    DataParallel necessitates objects to be gathered into GPU:0 to have 
+    dimension 0. 
+    This function will be used for scalar outputs of model forwards 
+    such as loss and n_exs.
+    """
+    if isinstance(cuda_devices, list):
+        if isinstance(obj, torch.Tensor) is False:
+            obj = torch.tensor(obj).cuda()
+        return obj.unsqueeze(0)
+    else:
+        return obj
```

Comment:
why don't we need to call `.cuda()` if we're using a single GPU?

Reply:
Because out["loss"] is already on cuda. We just explicitly put cuda here because we're creating a torch.Tensor.

Reply:
I guess my concern is making sure that we're not trying to call `torch.tensor` on something that shouldn't be a tensor ?

Reply:
The only case where we make it a tensor is for nn.DataParallel (when there are multiple gpu devices being used).